code/modules/power/supermatter/supermatter.dm

```diff
@@ -1025,6 +1038,26 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	COOLDOWN_START(src, common_radio_cooldown, SUPERMATTER_COMMON_RADIO_DELAY)
 	return TRUE
 
+/obj/machinery/power/supermatter_crystal/proc/holiday_lights()
+	holiday_lights = TRUE
+	RegisterSignal(src, COMSIG_ATOM_ITEM_INTERACTION, PROC_REF(holiday_attackby))
+	update_appearance()
+
+/// Consume the santa hat and add it as an overlay
+/obj/machinery/power/supermatter_crystal/proc/holiday_attackby(source, mob/living/user, obj/item/item, list/modifiers)
+	SIGNAL_HANDLER
+	if(istype(item, /obj/item/clothing/head/costume/santa))
+		QDEL_NULL(item)
+		RegisterSignal(src, COMSIG_ATOM_EXAMINE, PROC_REF(holiday_hat_examine))
+		add_overlay(mutable_appearance(icon, "santa_hat"))
```

Comment:
what if i slap the mini crystal with a santa hat

Reply:
okay we'll give the shard a hat too